internal/backends/winit/lib.rs

```diff
@@ -828,6 +831,14 @@ pub trait WinitWindowAccessor: private::WinitWindowAccessorSealed {
         callback: impl FnMut(&i_slint_core::api::Window, &winit::event::WindowEvent) -> WinitWindowEventResult
             + 'static,
     );
+
+    /// Returns a option of a future that resolves to the [`winit::window::Window`] for this Slint window.
+    /// If the window is not backed by the winit backend, this returns `None` immediately. When the future is
+    /// ready, the output it resolves to is either `Ok(Arc<winit::window::Window>)` if the window exists,
+    /// or an error if the window has been deleted in the meanwhile.
+    fn winit_window(
```

Comment:
This has a complex function signature. Have you considered async function is trait?
Then you'd return an error instead of None. But I have the feeling it'd be easier to use and more idiomatic. Perhaps an example in the doc would help

Reply:
I was contemplating that. What mainly makes me consider this over async is that when the window is not backed by the winit backend, we can return None immediately, without the need to poll the future (which implies a delay).

But if that’s not worth it, then async would indeed make for a much simpler signature.

Reply:
Is there something user can do in that case? (If None is returned.) Fallback to something else?

Reply:
Not really, I think it’s pretty binary. Give our singular backend nature this is - basically - only the case if you forget to explicitly select the backend.

I’ll simplify the signature to be an async fn and return a Result with a PlatformError.

Reply:
Had to keep the signature without `async` because of the warning about it for public traits, but I've removed the option and added a docs example.